compose/src/main/java/com/bottlerocketstudios/compose/snippets/CreateSnippet.kt

```diff
@@ -19,4 +60,168 @@ data class CreateSnippetScreenState(
+            onChanged = uiState.onContentsChanged,
+            hint = stringResource(id = R.string.create_snippet_contents_hint),
+            minLines = 5,
+            imeAction = ImeAction.Done
+        )
+        LabelledCheckbox(uiState = uiState)
+        FailedText(uiState = uiState)
+        CreateSnippetButton(uiState = uiState)
+    }
+}
+
+@Composable
+fun CreateSnippetButton(uiState: CreateSnippetScreenState) {
+    val enabled = uiState.createEnabled.value
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+    ) {
+        TextButton(
+            onClick = { uiState.onCreateClicked() },
+            enabled = enabled,
+            colors = ButtonDefaults.textButtonColors(backgroundColor = ArchitectureDemoTheme.colors.primary),
+            modifier = Modifier
+                .padding(top = Dimens.grid_0_5)
+                .defaultMinSize(minWidth = Dimens.grid_12_5)
+        ) {
+            Text(
+                stringResource(id = R.string.create_snippet_button).uppercase(),
+                color = ArchitectureDemoTheme.colors.onPrimary,
+                style = MaterialTheme.typography.h3.bold()
+            )
+        }
+    }
+}
+
+@Composable
+fun FailedText(uiState: CreateSnippetScreenState) {
+    val text = if (uiState.failed.value) {
+        stringResource(id = R.string.snippet_creation_failed)
+    } else {
+        ""
+    }
+    Text(
+        text = text,
+        textAlign = TextAlign.Center,
+        style = MaterialTheme.typography.h4.normal(),
+        modifier = Modifier
+            .padding(
+                start = Dimens.grid_7,
+                end = Dimens.grid_7,
+                top = Dimens.grid_1,
+                bottom = Dimens.grid_1
+            )
+            .fillMaxWidth()
+            .wrapContentHeight()
+    )
+}
+
+@Composable
+fun LabelledCheckbox(uiState: CreateSnippetScreenState) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .padding(
+                start = Dimens.grid_5,
+                end = Dimens.grid_5,
+            )
+    ) {
+        Checkbox(
+            checked = uiState.private.value,
+            onCheckedChange = { uiState.onPrivateChanged(it) },
+            enabled = true,
+            colors = CheckboxDefaults.colors(ArchitectureDemoTheme.colors.tertiary)
+        )
+        Text(text = stringResource(id = R.string.create_snippet_private))
+    }
+}
+
+@Composable
+fun TextInputBox(
+    text: String,
+    onChanged: (String) -> Unit,
+    hint: String,
+    minLines: Int = 1,
+    topPadding: Dp = Dimens.grid_1,
+    bottomPadding: Dp = Dimens.grid_1,
+    imeAction: ImeAction = ImeAction.Next
+) {
+    val focusManager = LocalFocusManager.current
+    val lineHeight = MaterialTheme.typography.h2.fontSize * 4 / 3
```

Comment:
@VanyaHuamanBottleRocket Is this a common practice in compose (adjusting lineHeight based on the font size)? If so, it seems like it might be slightly more readable to use `* 1.333` instead of `* 4 / 3`

cc: @BottleRocket-Colin

Reply:
No this is not common. I could not find another way of putting a min number of lines that would automatically expand the view to the height of said lines. I thought of putting a height strictly based on DP but it wouldn't account for the different text size.

Reply:
After talking with Colin I am going to remove this and just use a minDefault Height.